frontend/src/app/chatmenu/BlockUser.tsx

```diff
@@ -1,66 +1,81 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import './BlockUser.css'
-import '../ui/User.css'
 
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
-import { faUnlock, faUserLock } from '@fortawesome/free-solid-svg-icons'
+import { faLock, faLockOpen } from '@fortawesome/free-solid-svg-icons'
 
 import { ChatMenuPageProps } from './ChatMenuPage'
 import { UserData } from '../types/UserData'
+import { Button, UserTile } from '../ui'
+import { fetchFromBackend, useBackendFetchJson } from '../utils'
 
 interface BlockUserProps {
     participants: UserData[]
+    myUid: string
 }
 
-// create user tile to display participants
-const UserTile = ({ user }) => {
-    const { uid, displayName, photoURL } = user
-    const alt = `Profile image for ${displayName}.`
+function useFetchBlockedUsers(myUid: string) {
+    const [res] = useBackendFetchJson({
+        route: `/chat/block/user/${myUid}`,
+        deps: [myUid],
+        isValid: myUid,
+    })
+    const blockedUsers = res?.blocks || []
+    // create a blocked user dictionary
+    return blockedUsers.reduce(
+        (agg, user: { uid: string; blocked: boolean }) => ({
+            ...agg,
+            [user.uid]: user.blocked,
+        }),
+        {}
+    )
+}
 
-    const doBlock = () => {
-        // TODO: Implement blocking the user method
-    }
+async function updateBlock(blockUid: string, shouldBlock: boolean) {
+    await fetchFromBackend({
+        route: `/chat/block/user/${this?.BlockUserProps?.myUid}/${blockUid}/${shouldBlock}`,
```

Comment:
Ah, this is why the block results don't get saved. I don't think there is a problem with the `useEffect()`, instead the problem is that we send an `undefined` value for the `from_uid` in the endpoint, so there is no way to know who is blocking this user.

The reason is that `this` does not have access to the properties we are using here.

There are a few different wants to handle this:

Option A: Revise the `updateBlock()` function to take `myUid` as an input, which means that we then have to pass `myUid` into the `CreateUserTile` component. The downside of this is that now we are drilling a prop we shouldn't really need.

Option B: Pass an `updateBlock()` function to the `CreateUserTile` component that takes only `blockUid` and `shouldBlock` as arguments, then use partial application to create a function that already has `myUid` set, then pass that partially applied function to the component. This might add more indirection, but it limits what the inner component needs to provide.

In the outer component, it would look something like this:

```jsx
async function blockUser(myUid, blockUid, updateBlock ) { ... }

// Renamed from CreateUserTile
function BlockUserTile({ user, value, submitBlock }) {
  const blockUid = user?.uid
  const [blocked, setIsBlocked] = useState(value)

  const blockHandler = () => {
    const newValue = !value
    submitBlock(blockUid, newValue)
    setIsBlocked(newValue)
  }

  return (...)
}

function OuterComponent({ myUid }) {
  // Partial application turns a function that needs three inputs into one that needs only two
  const updateBlock = (blockUid, shouldBlock) => blockUser(myUid, blockUid, shouldBlock)
  return (
    ...
    <BlockUserTile user={user} value={value} updateBlock={updateBlock} />
  )
}
```